File: byte-buddy-gradle-plugin/android-plugin-test/src/test/assets/dummyProjects/jar_targeting_libraries/java/BasicLibTransformation.java
import net.bytebuddy.build.Plugin;
import net.bytebuddy.description.type.TypeDescription;
import net.bytebuddy.dynamic.ClassFileLocator;
import net.bytebuddy.dynamic.DynamicType;
import net.bytebuddy.implementation.FixedValue;
import net.bytebuddy.implementation.MethodDelegation;
import net.bytebuddy.matcher.ElementMatchers;

import java.io.IOException;

public class BasicLibTransformation implements Plugin {

    @Override
    public DynamicType.Builder<?> apply(DynamicType.Builder<?> builder,
                                        TypeDescription typeDescription,
                                        ClassFileLocator classFileLocator) {
        return builder.method(ElementMatchers.named("getMessage"))
                .intercept(FixedValue.value("Instrumented message in lib"));
    }

    @Override
    public boolean matches(TypeDescription target) {
        return target.getName().equals("SomeClass") || target.getName().equals("SomeLibClass");
    }

    @Override
    public void close() throws IOException {

    }
}

File: byte-buddy-gradle-plugin/android-plugin-test/src/test/assets/dummyProjects/jar_with_kotlin/java/BasicLibTransformation.kt
import net.bytebuddy.build.Plugin
import net.bytebuddy.description.type.TypeDescription
import net.bytebuddy.dynamic.ClassFileLocator
import net.bytebuddy.dynamic.DynamicType
import net.bytebuddy.implementation.FixedValue
import net.bytebuddy.matcher.ElementMatchers
import net.bytebuddy.description.method.MethodDescription

class BasicLibTransformation : Plugin {

    override fun matches(target: TypeDescription?): Boolean {
        return target?.getName()?.contains("SomeClass") == true || target?.getName()
            ?.contains("SomeKotlinClass") == true
    }

    override fun apply(
        builder: DynamicType.Builder<*>,
        typeDescription: TypeDescription,
        classFileLocator: ClassFileLocator
    ): DynamicType.Builder<*> {
//        val assignableTo: Boolean = typeDescription.isAssignableTo(String::class.java)
//        println("Is $typeDescription assignable2: $assignableTo")
        return builder.method(
            ElementMatchers.named<MethodDescription>(
                "getMessage"
            )
        ).intercept(FixedValue.value("Instrumented message in lib"))
    }

    override fun close() {
        // Nothing to close
    }
}